cluster-autoscaler/cloudprovider/clusterapi/clusterapi_utils.go

```diff
@@ -407,3 +415,25 @@ func GetDefaultScaleFromZeroArchitecture() SystemArchitecture {
 	})
 	return *systemArchitecture
 }
+
+// getManagedNodeLabelsFromLabels returns a map of labels that will be propagated
+// to nodes based on the Cluster API metadata propagation rules.
+func getManagedNodeLabelsFromLabels(labels map[string]string) map[string]string {
+	// TODO elmiko, add a user configuration to inject a string with their `--additional-sync-machine-labels` string.
+	// ref: https://cluster-api.sigs.k8s.io/reference/api/metadata-propagation#machine
+	managedLabels := map[string]string{}
+	for key, value := range labels {
+		dnsSubdomainOrName := strings.Split(key, "/")[0]
```

Comment:
Should we split this out into a separate func?

```golang
for key, value := range labels {
    if isManagedLabel(key) {
        managedLabels[key] = value
    }
}
```

Or maybe something else to avoid repeating the identical `managedLabels[key] = value` assignments.

Reply:
sure thing, good suggestion!